osler/core/models.py

```diff
@@ -536,3 +566,48 @@ def admin_url(self):
     def __str__(self):
         return " ".join(["AI for", str(self.patient) + ":",
                          str(self.instruction), "due on", str(self.due_date)])
+
+
+class EncounterStatus(models.Model):
+    '''Different status for encounter, as simple as Active/Inactive 
+    or Waiting/Team in Room/Attending etc'''
+    name = models.CharField(max_length=100, primary_key=True)
+    is_active = models.BooleanField(default=False)
+
+    def __str__(self):
+        return self.name
+
+
+def default_active_status():
+    status, created = EncounterStatus.objects.get_or_create(
+        name=settings.OSLER_DEFAULT_ACTIVE_STATUS[0],
+        is_active=settings.OSLER_DEFAULT_ACTIVE_STATUS[1])
+    return status
+
+
+def default_inactive_status():
+    status, created = EncounterStatus.objects.get_or_create(
+        name=settings.OSLER_DEFAULT_INACTIVE_STATUS[0],
+        is_active=settings.OSLER_DEFAULT_INACTIVE_STATUS[1])
+    return status
+
+
+class Encounter(SortableMixin):
+    '''Encounter for a given patient on a given clinic day
+    Holds all associated notes, labs, etc performed on that clinic day
+    Can reoder in admin panel for Active Patients page'''
+    class Meta:
+        ordering = ['order']
+    
+    order = models.PositiveIntegerField(default=0, editable=False, db_index=True)
+    patient = models.ForeignKey(Patient, on_delete=models.PROTECT)
+    clinic_day = models.DateField()
+    status = models.ForeignKey(EncounterStatus, on_delete=models.PROTECT)
+
+    sorting_filters = (
+        ('Active Encounters', {'status__is_active': True}),
+        )
+
+    def __str__(self):
+        return str(self.patient)+" on "+datetime.datetime.strftime(self.clinic_day, '%A, %B %d, %Y')
```

Comment:
Make sure this will be compatible with internationalization

Reply:
This is just making a datetime into a string, clinic_day is saved using a timezone-aware function. I copied this from.... the str for ClinicDate I think

Reply:
Yeah I think this is fine, since `clinic_day` points to a DateField (which should be tz aware).

However, we can also do `clinic_day.strftime()`, so I've updated it to do it that way!